contentcuration/contentcuration/frontend/shared/data/serverSync.js

```diff
@@ -353,23 +399,71 @@ let intervalTimer;
 export function startSyncing() {
   // Initiate a sync immediately in case any data
   // is left over in the database.
-
+  debouncedSyncChanges();
   const websocketUrl = new URL(
     `/ws/sync_socket/${window.CHANNEL_EDIT_GLOBAL.channel_id}/`,
     window.location.href
   );
   websocketUrl.protocol = window.location.protocol == 'https:' ? 'wss:' : 'ws:';
+
   socket = new WebSocket(websocketUrl);
 
   // Connection opened
   socket.addEventListener('open', () => {
     console.log('Websocket connected');
   });
 
-  debouncedSyncChanges();
+  // Listen for any errors due to which connection may be closed.
+  socket.addEventListener('error', event => {
+    console.log('WebSocket error: ', event);
+  });
+
+  socket.addEventListener('message', e => {
+    const data = JSON.parse(e.data);
+    console.log(data);
+    if (data.task) {
+      Task.setTasks([data.task]);
+    }
+    if (data.response_payload && data.response_payload.allowed) {
+      try {
+        handleAllowed(data.response_payload.allowed);
+      } catch (err) {
+        console.log(err);
+      }
+    }
+    if (data.response_payload && data.response_payload.disallowed) {
+      try {
+        handleDisallowed(data.response_payload.disallowed);
+      } catch (err) {
+        console.log(err);
+      }
+    }
+    if (data.change) {
+      try {
+        Promise.all([
+          handleReturnedChanges([data.change]),
+          handleMaxRevs([data.change], data.change.created_by_id),
+        ]);
```

Comment:
Wrapping these two in `Promise.all([ ... ])` is unnecessary here, since nothing is waiting on that promise.